src/Storages/ObjectStorage/StorageObjectStorageSource.cpp

```diff
@@ -430,16 +431,32 @@ StorageObjectStorageSource::ReaderHolder StorageObjectStorageSource::createReade
     ObjectInfoPtr object_info;
     auto query_settings = configuration->getQuerySettings(context_);
 
+    bool not_a_path = false;
+
     do
     {
+        not_a_path = false;
         object_info = file_iterator->next(processor);
 
         if (!object_info || object_info->getPath().empty())
             return {};
 
+        StorageObjectStorageStableTaskDistributor::CommandInTaskResponse command(object_info->getPath());
+        if (command.is_parsed())
```

Comment:
Not elegant, but for backward compatibility left string as filename and JSON as something with additional information.
Can be extended in future.

Reply:
It looks like storing a JSON in a variable named as PATH, which is highly misleading... If you want to pass some extra data as JSON, please do it explicitly.

Either by storing some extra information in `ObjectInfo`'s `ObjectMetadata`, by updating `ObjectMetadata` with extra field (preferred),  OR inserting a special key into `ObjectMetadata::attributes` with stringified JSON (less desirable), OR doing something else.